### PR TITLE
introduce permissions for User management

### PR DESCRIPTION
**Pull request by Bartzi**
contributes to #5 
introduces new way to work with permissions on object level
I've implemented the base for further work ;)
it even includes an editor for changing permissions of documents
Documents are now polymorphic models

### CODE REVIEW

File: _1327/documents/migrations/0003_auto_20150216_1744.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('documents', '0002_auto_20141101_2244'),
+    ]
+
+    operations = [
+        migrations.AlterModelOptions(
+            name='document',
+            options={'permissions': (('view_document', 'User/Group is allowed to View that Document'),)},
```
Comment (janno42) [inline _1327/documents/migrations/0003_auto_20150216_1744.py]: ... to view...

File: _1327/documents/models.py [hunk 2]
```diff
@@ -2,25 +2,24 @@
...
 	text = models.TextField()
-	type = models.CharField(max_length=5, default='I')
 
-	types = (
-		('I', _('Information')),
-		('P', _('Protocol')),
-	)
+	class Meta:
+		permissions = (
+			('view_document', 'User/Group is allowed to View that Document'),
```
Comment (janno42) [inline _1327/documents/models.py]: ... to view...

**Comment by invliD** [comment]
tl;dr

File: _1327/documents/forms.py [hunk 3]
```diff
@@ -18,5 +18,40 @@ class TextForm(forms.Form):
...
+				if self.cleaned_data["change_permission"]:
+					assign_perm(permission.codename, group, model)
+				else:
+					remove_perm(permission.codename, group, model)
+			elif "delete" in str(permission):
+				if self.cleaned_data["delete_permission"]:
+					assign_perm(permission.codename, group, model)
+				else:
+					remove_perm(permission.codename, group, model)
+			elif "View" in str(permission):
```
Comment (janno42) [inline _1327/documents/forms.py]: view

File: _1327/information_pages/templates/information_pages_base.html [hunk 4]
```diff
@@ -14,15 +14,19 @@
 	</div>
 	{% block information_page_nav %}
 		{% if request.user.is_staff %}
-			<div class="divider"></div>
```
Comment (janno42) [inline _1327/information_pages/templates/information_pages_base.html]: keep this line and indentation below

File: _1327/information_pages/templates/information_pages_base.html [hunk 5]
```diff
@@ -14,15 +14,19 @@
...
+		<a class="btn btn-sm btn-sidebar {% if active_page == 'view' %}btn-warning{% else %}btn-default{% endif %}" href="{% url 'information_pages:view_information' document.url_title %}" role="tab">{% trans "View" %}</a>
+		<a class="btn btn-sm btn-sidebar {% if active_page == 'edit' %}btn-warning{% else %}btn-default{% endif %}" href="{% url 'information_pages:edit' document.url_title %}" role="tab">{% trans "Edit" %}</a>
+		<a class="btn btn-sm btn-sidebar {% if active_page == 'permissions' %}btn-warning{% else %}btn-default{% endif %}" href="{% url 'information_pages:permissions' document.url_title %}" role="tab">{% trans "Permissions" %}</a>
+		<a class="btn btn-sm btn-sidebar {% if active_page == 'versions' %}btn-warning{% else %}btn-default{% endif %}" href="{% url 'information_pages:versions' document.url_title %}" role="tab">{% trans "Versions" %}</a>
 		{% endif %}
 	{% endblock %}
 {% endblock %}
 
 {% block content %}	
+	<div class="meta-information">
```
Comment (janno42) [inline _1327/information_pages/templates/information_pages_base.html]: remove these 4 lines

**Comment by janno42** [comment]
:+1:

**Comment by janno42** [comment]
:shipit:

**Comment by janno42** [comment]
:ship:
